src/sources/http_client/client.rs

```diff
@@ -307,8 +307,7 @@ impl http_client::HttpClientContext for HttpClientContext {
     fn on_response(&mut self, _url: &Uri, _header: &Parts, body: &Bytes) -> Option<Vec<Event>> {
         // get the body into a byte array
         let mut buf = BytesMut::new();
-        let body = String::from_utf8_lossy(body);
-        buf.extend_from_slice(body.as_bytes());
+        buf.extend_from_slice(body);
```

Comment:
Is this changing the behavior for all of the other codecs? They previously always received the lossy conversion, now only `json` does.

Reply:
Yes, you are correct. I took a quick scan over the existing deserializers and `Syslog`, `NativeJson`, and `Gelf` will no longer apply a lossy conversion, which is undesirable. Conversely, lossy conversion is no longer applied to the input of `Bytes` and `Native` which is desired.

Do you have any suggestions for a way forward? We could add the conversion into the Deserializers itself, similar to what we did for the json deserializer.

Reply:
Could we do something along the lines of: 

```Rust
match self.decoder.deserializer {
  // apply lossy conversion to Syslog, NativeJson, Gelf
}
```

?

Reply:
That does seem like the easiest way to slide this change in without additional updates to other codecs. Longer term it does seem appropriate to standardize more on the lossy behavior, but I don't know what the priority/capacity is for that.

Reply:
Ah, yeah, I neglected that other codecs also need UTF-8 aside from the `json` codec when thinking about this issue. I think ideally all codecs would grow the new lossy option. I'd support doing that now since it doesn't seem like a _huge_ lift based on https://github.com/vectordotdev/vector/pull/17628, but, if it is, I'd support the workaround mentioned by Kyle.

Reply:
😄 sorry for making more work for you @dsmith3197

Reply:
No worries, I'll go ahead with adding the lossy option to the relevant codecs 😄

Reply:
This has now been resolved thanks to https://github.com/vectordotdev/vector/pull/17688.